### PR TITLE
hw20, JDBC + Templates

### PR DESCRIPTION
**Pull request by u2107680007911543774e7r**

### CODE REVIEW

File: src/main/java/lesson_jdbc_01/CustomerDaoImplGeneric.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.sql.SQLException;
+
+public class CustomerDaoImplGeneric {
+
+	public boolean insertCustomer(Customer customer) throws SQLException {
+		return new TemplateAbstract<Customer>() {
+			@Override
+			public PreparedStatement returnPreparedStatement(Customer customer, Connection conn) throws SQLException {
+				PreparedStatement stmt;
+				conn = ConnectToDB.getConnection();
```
Comment (Cherkashyn) [inline src/main/java/lesson_jdbc_01/CustomerDaoImplGeneric.java]: Is it necessary to get connection in each method? You can pass connection as parameter.

**Review by Cherkashyn — APPROVED** [review]
Good and short.

File: pom.xml [hunk 2]
```diff
@@ -21,9 +21,30 @@
...
+		<spring.version>4.1.6.RELEASE</spring.version>
+		<dbh2.version>1.4.187</dbh2.version>
+		<derby.version>10.11.1.1</derby.version>
 	</properties>
 
 	<dependencies>
+		<!-- Spring JDBC -->
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-jdbc</artifactId>
```
Comment (SerhiiPasko) [inline pom.xml:33]: Spring don't need for you...

File: src/main/java/lesson_jdbc_01/CustomerDaoImplGeneric.java (hunk 1 above)
Comment (SerhiiPasko) [reply src/main/java/lesson_jdbc_01/CustomerDaoImplGeneric.java]: Yes, better to template

**Review by SerhiiPasko — APPROVED** [review]
You didn't like to use generics. It possible in this case, good...

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Extends DaoInterface it critical

File: src/main/java/lesson_jdbc_01/CustomerDaoImplInterface.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package lesson_jdbc_01;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public class CustomerDaoImplInterface implements CustomerDaoInterface{
```
Comment (SerhiiPasko) [inline src/main/java/lesson_jdbc_01/CustomerDaoImplInterface.java:6]: Crazy names

**Review by SerhiiPasko — APPROVED** [review]
Good. Read comments

**Pull request merged**